### PR TITLE
Fixed transform fucntion to replace the object

### PR DESCRIPTION
**Pull request by ashayshirwadkar**
Previously the transformed object was overwritten to the old object.
But the logic failed if transformed object is less in size compared
to original object. Therefore, trucating the original object is
necessary.

### CODE REVIEW

File: src/cls/tabular/cls_tabular.cc [hunk 1]
```diff
@@ -1997,8 +1987,8 @@ int transform_db_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
         delete meta_builder;
 
         // Write the object back to Ceph
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular.cc]: add comment that replace is truncate and write full

File: src/cls/tabular/cls_tabular.cc [hunk 2]
```diff
@@ -1908,17 +1907,8 @@ int transform_db_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
...
-    if (format_type == op.required_type) {
-        // Source and destination object types are same, therefore no tranformation
-        // is required.
-        CLS_LOG(20, "No Transforming required");
-        return 0;
-    }
-
     // Object is sequence of actual data along with encoded metadata
     bufferlist encoded_meta_bls;
     int ret = cls_cxx_read(hctx, 0, 0, &encoded_meta_bls);
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular.cc:1916]: add TODO: get individ off/len of fbmeta's inside obj and read one at a time.

Comment (ashayshirwadkar) [reply src/cls/tabular/cls_tabular.cc]: Done

**Review by jlefevre — APPROVED** [review]
LGTM

**Pull request merged**